spec/core/config_mixin_spec.rb

```diff
@@ -0,0 +1,73 @@
+require "logstash/config/mixin"
+require "logstash/filters/base"
+
+describe LogStash::Config::Mixin do
+  context "when validating :bytes" do
+    it "should validate :bytes" do
+      conf_class = Class.new(LogStash::Filters::Base) do
+        include LogStash::Config::Mixin
+        config_name "test"
+        milestone 1
+        config :size_file, :validate => :bytes
+      end.new({"size_file" => "10Kib"})
+      expect(conf_class.size_file).to eq(10240)
+    end
+
+    it "should allow setting valid default" do
+      conf_class = Class.new(LogStash::Filters::Base) do
+        include LogStash::Config::Mixin
+        config_name "test"
+        milestone 1
+        config :size_file, :validate => :bytes, :default => "10Kib"
+      end.new({})
+      expect(conf_class.size_file).to eq(10240)
+    end
+
+    it "should accept bytes with no units specified in suffix" do
+      conf_class = Class.new(LogStash::Filters::Base) do
+        include LogStash::Config::Mixin
+        config_name "test"
+        milestone 1
+        config :size_file, :validate => :bytes, :default => "1024"
+      end.new({})
+      expect(conf_class.size_file).to eq(1024)
+    end
+
+    it "should be case-insensitive" do
+      conf_class = Class.new(LogStash::Filters::Base) do
+        include LogStash::Config::Mixin
+        config_name "test"
+        milestone 1
+        config :size_file_1, :validate => :bytes, :default => "10KIB"
+        config :size_file_2, :validate => :bytes, :default => "10kib"
+      end.new({})
+      expected_bytes = 10240
+      expect(conf_class.size_file_1).to eq(expected_bytes)
+      expect(conf_class.size_file_2).to eq(expected_bytes)
+    end
+
+    it "should be space-insensitive" do
+      conf_class = Class.new(LogStash::Filters::Base) do
```

Comment:
I wonder if we could shorten these tests...

Have one subject with all config options to test:
```ruby
let(:number) { rand(1000) }
subject do
  Class.new(...) do
    config :size_capitalized_default, :validate => :bytes, :default => "#{number}KIB".upcase
    config :size_lowercase_default, :validate => :bytes, :default => "#{number}KIB".downcase
    ... other configs here ...
  end.new({})
end

# Then short `it` clauses that test individual cases with the same subject
it "should be case insensitive" do
  expect(subject.size_capitalized_default).to(eq(subject.size_lowercase_default))
  expect(subject.size_capitalized_default).to(eq(number *1024))
end
```